roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+            self.register_failure(
+                'No nodes appear to be defined according to the API.'
+            )
+
+        for node in nodes:
+            self.check_node_kubelet(node)
+
+    def get_resource(self, kind):
+        """Return a list of all resources of the specified kind."""
+        for resource in self.task_vars['resources']['results']:
+            if resource['item'] == kind:
+                return resource['results']['results'][0]['items']
+
+        raise OpenShiftCheckException('CouldNotListResource',
+                                      'Could not list resource %s' % kind)
+
+    def check_node(self):
+        """Gather diagnostic information on a node and perform connectivity
+        checks on pods and services."""
+        node_name = self.get_node_name()
+        if not node_name:
+            self.register_failure(
+                'Could not find node name in node-config.yaml.'
+            )
+            return
+
+        if self.want_full_results:
+            try:
+                # XXX: Loading node logs can take in inordinate amount of time
+                # on busy systems.
+                # service_prefix = self.get_var('openshift_service_type')
+                # if self._templar is not None:
+                #     service_prefix = self._templar.template(service_prefix)
+                # self.save_service_logs('%s-node' % service_prefix)
+                self.save_command_output('bridges', ['brctl', 'show'])
+
+                self.save_command_output('docker-ps', ['docker', 'ps'])
+
+                self.save_command_output('flows', ['ovs-ofctl',
+                                                   '-O', 'OpenFlow13',
+                                                   'dump-flows', 'br0'])
+                self.save_command_output('ovs-show', ['ovs-ofctl',
+                                                      '-O', 'OpenFlow13',
+                                                      'show', 'br0'])
+
+                self.save_command_output('tc-qdisc', ['tc', 'qdisc', 'show'])
+                self.save_command_output('tc-class', ['tc', 'class', 'show'])
+                self.save_command_output('tc-filter', ['tc', 'filter', 'show'])
+
+                docker_unit = self.read_command_output(['systemctl', 'cat',
+                                                        'docker.service'])
+                self.register_file('docker-unit-file', docker_unit)
+
+                # Save docker service unit environment file for openshift-sdn,
+                # if any.
+                #
+                # The systemd and service modules both return the unit as a
+                # dict with only the last value for a given key, and there may
+                # be multiple EnvironmentFile stanzas, so we need to slurp and
+                # parse the service unit without the aid of these two modules.
+                #
+                # XXX: Do we really need to look for an 'openshift-sdn'
```

Comment:
This was needed before we did the CNI change, so you can probably axe it.

@dcbw am I remembering correctly?